app/controllers/spree/admin/products_controller_decorator.rb

```diff
@@ -92,9 +94,28 @@ def collection_actions
 
   private
 
+  def latest_import
+    @show_latest_import = params[:latest_import] || false
+  end
+
   def load_form_data
     @producers = OpenFoodNetwork::Permissions.new(spree_current_user).managed_product_enterprises.is_primary_producer.by_name
     @taxons = Spree::Taxon.order(:name)
+    import_dates = [{id: '0', name: ''}]
+    product_import_dates.map { |i| import_dates.push(id: i.to_date, name: i.to_date.to_formatted_s(:long)) }
+    @import_dates = import_dates.uniq.to_json
+  end
+
+  def product_import_dates
+    import_dates = Spree::Variant.
+      select('spree_variants.import_date').
+      joins(:product).
+      where('spree_products.supplier_id IN (?)
+      AND spree_variants.is_master = false
+      AND spree_variants.import_date IS NOT NULL
```

Comment:
do you know you can have multiple calls to `where`? they will be appended to the query through ANDs. This way you can use a string just in the constraint above and have all the others with a hash.